lib/march_hare/session.rb

```diff
@@ -583,6 +588,7 @@ def converting_rjc_exceptions_to_ruby(&block)
     def reconnecting_on_network_failures(interval_in_ms, &fn)
       @logger.debug("session: reconnecting_on_network_failures")
       begin
+        return if @was_explicitly_closed
```

Comment:
Somehow seems better to have the loop exit in there instead of in the rescue, but it's basically the same